src/map.h

```diff
@@ -874,6 +874,7 @@ void add_corpse( const tripoint &p );
 
 // mapgen.cpp functions
  void generate(const int x, const int y, const int z, const int turn);
+ void generate_uniform(const int x, const int y, const int z, const oter_id &terrain_type);
```

Comment:
And this could be a global function not exposed in any header. It is not used outside of map.cpp (move it there, or declare it there), and it certainly does not use any members of the map class.